src/bika/lims/validators.py

```diff
@@ -1550,6 +1551,55 @@ def __call__(self, value, **kwargs):
             ))
 
 
+class UniqueReferenceSampleIDValidator(object):
+    """
+    Ensures that no existing object in the system has an ID matching the value
+    of the field to which this validator is applied.
+    """
+    implements(IValidator)
+    name = "unique_referencesample_id_validator"
+
+    def __call__(self, value, **kwargs):
+        instance = kwargs["instance"]
+
+        # skip if no value provided
+        if not value:
+            return
+
+        # skip if the value matches with object's current id
+        if instance.getId() == value:
+            return
+
+        # check if the id is valid
+        parent = api.get_parent(instance)
+        if not api.is_valid_id(value, container=parent):
```

Comment:
Ah, here it comes.
So I guess the fallback from above is just to be sure?
In that case, I would prefer to raise a `ValueError` instead and only do the fallback if no `ManualID` is set.

Reply:
Yes, done with 21e7970d9c9a711426c7b69547510dc7fcf4e99c